datalad/distribution/tests/test_siblings.py

```diff
@@ -0,0 +1,175 @@
+    with chpwd(repo_path):  # not yet there
+        assert_raises(InsufficientArgumentsError,
+                      siblings, 'add', url=httpurl1)
+
+    # prepare src
+    source = install(repo_path, source=origin, recursive=True)
+    # pollute config
+    depvar = 'remote.test-remote.datalad-publish-depends'
+    source.config.add(depvar, 'stupid', where='local')
+
+    # cannot configure unknown remotes as dependencies
+    res = siblings(
+        'configure',
+        dataset=source,
+        name="test-remote",
+        url=httpurl1,
+        publish_depends=['r1', 'r2'],
+        on_failure='ignore')
+    assert_status('error', res)
+    assert_in('unknown sibling(s) specified as publication dependency',
+              res[0]['message'])
+    # prior config was changed by failed call above
+    eq_(source.config.get(depvar, None), 'stupid')
+
+    res = siblings('configure',
+                   dataset=source, name="test-remote",
+                   url=httpurl1,
+                   result_xfm='paths')
+
+    eq_(res, [source.path])
+    assert_in("test-remote", source.repo.get_remotes())
+    eq_(httpurl1,
+        source.repo.get_remote_url("test-remote"))
+
+    # reconfiguring doesn't change anything
+    siblings('configure', dataset=source, name="test-remote",
+             url=httpurl1)
+    assert_in("test-remote", source.repo.get_remotes())
+    eq_(httpurl1,
+        source.repo.get_remote_url("test-remote"))
+    # re-adding doesn't work
+    res = siblings('add', dataset=source, name="test-remote",
+                   url=httpurl1, on_failure='ignore')
+    assert_status('error', res)
+    # only after removal
+    res = siblings('remove', dataset=source, name="test-remote")
+    assert_status('ok', res)
+    assert_not_in("test-remote", source.repo.get_remotes())
+    res = siblings('add', dataset=source, name="test-remote",
+                   url=httpurl1, on_failure='ignore')
+    assert_status('ok', res)
+
+    # add to another remote automagically taking it from the url
+    # and being in the dataset directory
+    with chpwd(source.path):
+        res = siblings('add', url=httpurl2)
+    assert_result_count(
+        res, 1,
+        name="remote2.example.com", type='sibling')
+    assert_in("remote2.example.com", source.repo.get_remotes())
+
+    # don't fail with conflicting url, when using force:
+    res = siblings('configure',
+                    dataset=source, name="test-remote",
+                    url=httpurl1 + "/elsewhere")
+    assert_status('ok', res)
+    eq_(httpurl1 + "/elsewhere",
+        source.repo.get_remote_url("test-remote"))
+
+
+    # no longer a use case, I would need additional convincing that
+    # this is anyhow useful other then tripple checking other peoples
+    # errors. for an actual check use 'query'
+    # maybe it could be turned into a set of warnings when `configure`
+    # alters an existing setting, but then why call configure, if you
+    # want to keep the old values
```

Comment:
Yes for 'warnings', yes for 'stop doing it', yes for 'why calling configure at all'?